src/single-spa-react.js

```diff
@@ -248,3 +239,72 @@ function reactDomRender({ opts, elementToRender, domElement, whenFinished }) {
   // default to this if 'renderType' is null or doesn't match the other options
   return opts.ReactDOM.render(elementToRender, domElement, whenFinished);
 }
+
+function getElementToRender(opts, props) {
+  const rootComponentElement = opts.React.createElement(
+    opts.rootComponent,
+    props
+  );
+
+  let elementToRender = SingleSpaContext
+    ? opts.React.createElement(
+        SingleSpaContext.Provider,
+        { value: props },
+        rootComponentElement
+      )
+    : rootComponentElement;
+
+  if (opts.errorBoundary) {
+    elementToRender = opts.React.createElement(
+      createErrorBoundary(opts),
+      props,
+      elementToRender
+    );
+  }
+
+  return elementToRender;
+}
+
+function createErrorBoundary(opts) {
+  // Avoiding babel output for class syntax and super()
+  // to avoid bloat
+  function SingleSpaReactErrorBoundary(props) {
```

Comment:
This is interesting. Just to make sure I’m following, here you’re creating a JS class (which is actually a function) directly with a function to reduce bundle size?

What (if anything) does the normal ‘...extends react.component’ part of the class declaration give us that this is missing?

Reply:
The behavior is the same as using class syntax afaik. I'm simulating the super() call below

Reply:
Very fancy.